bin/regrade.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+"""
+Expected directory structure:
+<BASE_PATH>/courses/<SEMESTER>/<COURSES>/submissions/<HW>/<USERNAME>/<VERSION#>
+
+This script will find all submissions that match the provided
+pattern and add them to the grading queue.
+
+USAGE:
+    regrade.py  <one or more (absolute or relative) PATTERN PATH>
+    regrade.py  <one or more (absolute or relative) PATTERN PATH>  --interactive
+"""
+
+import argparse
+import json
+import os
+import sys
+import glob
+
+SUBMITTY_DATA_DIR = "__INSTALL__FILLIN__SUBMITTY_DATA_DIR__"
+
+
+def arg_parse():
+    parser = argparse.ArgumentParser(description="Re-adds any submission folders found in the given path and adds"
+                                                 "them to a queue (default batch) for regrading")
+    parser.add_argument("path", nargs="+", metavar="PATH", help="Path (absolute or relative) to submissions to regrade")
+    parser.add_argument("--interactive", dest="interactive", action='store_const', const=True, default=False,
+                        help="What queue (INTERACTIVE or BATCH) to use for the regrading. Default "
+                        "is batch.")
+    return parser.parse_args()
+
+
+def main():
+    args = arg_parse()
+    data_dir = os.path.join(SUBMITTY_DATA_DIR, "courses")
+    data_dirs = data_dir.split(os.sep)
+    grade_queue = []
+
+    for input_path in args.path:
+        # handle relative path
+        if input_path == '.':
+            input_path = os.getcwd()
+        if input_path[0] != '/':
+            input_path = os.getcwd() + '/' + input_path
+        # remove trailing slash (if any)
+        input_path = input_path.rstrip('/')
+        # split the path into directories
+        dirs = input_path.split(os.sep)
+
+        # must be in the known submitty base data directory
+        if dirs[0:len(data_dirs)] != data_dirs:
+            print("ERROR: BAD REGRADE SUBMISSIONS PATH",input_path)
+            raise SystemExit("You need to point to a directory within {}".format(data_dir))
+            continue
```

Comment:
don't need this `continue` as we're exiting on this error? unless we actually want to just skip this directory and continue on with the other inputs?